Sources/Testing/Running/EntryPoint.swift

```diff
@@ -0,0 +1,82 @@
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+//
+
+@_implementationOnly import TestingInternals
+
+#if !SWT_TARGET_OS_APPLE
+/// The entry point to the testing library used by Swift Package Manager.
+///
+/// - Returns: The result of invoking the testing library. The type of this
+///   value is subject to change.
+///
+/// This function examines the command-line arguments to the current process
+/// and then invokes available tests in the current process.
+///
+/// - Warning: This function is used by Swift Package Manager. Do not call it
+///   directly.
+@_spi(SwiftPackageManagerSupport)
+public func __swiftPMEntryPoint() async -> CInt {
```

Comment:
Thus far, all double-underscored `public` APIs I've seen have been for use by macro expanded code. I wonder if this should only have one underscore, to preserve that convention?

Or hmm… maybe it should just have no underscores at all, since it is already `@_spi`? I could understand including underscores with its former name (`__main`) to avoid confusion, but now that it has a more specific name that might no longer be necessary.

Reply:
Yeah I can drop the underscores since it's SPI. That's a good point. I didn't have the SPI annotation at first.